src/main/java/oshi/software/os/linux/LinuxFileSystem.java

```diff
@@ -71,10 +114,19 @@
             // FileStore toString starts with path, then a space, then name in
             // parentheses e.g., "/ (/dev/sda1)" and "/proc (proc)"
             String path = store.toString().replace(" (" + store.name() + ")", "");
-            // Exclude special directories
-            if (path.startsWith("/proc") || path.startsWith("/sys") || path.startsWith("/run") || path.equals("/dev")
-                    || path.equals("/dev/pts"))
-                continue;
+
+            // Exclude pseudo file systems
+            if (this.pseudofs.contains(store.name())) {

```

Comment:
I've used three nested if statements (and a private method) because tmpfs
is used for system paths (e.g. in Fedora, /dev/shm) and ram disks (e.g. in
Debian, /tmp), so, if I want only ram disks I need to exclude these paths,
which differs between distributions (e.g. in Fedora I have /dev/shm
meanwhile in Debian 7 I have /run/shm). It's ugly, and I don't think if it
is the correct way, because the df(1) command shows all tmpfs (and
devtmpfs) file systems mounted in the os. What do you think? Is a good
choice to rollback this commit or is a good way to continue with this
direction?

2016-05-10 6:17 GMT+02:00 dbwiddis <notifications@github.com>:

> In src/main/java/oshi/software/os/linux/LinuxFileSystem.java
> <https://github.com/dblock/oshi/pull/163#discussion_r62610241>:
>
> > @@ -71,10 +114,19 @@
> >              // FileStore toString starts with path, then a space, then name in
> >              // parentheses e.g., "/ (/dev/sda1)" and "/proc (proc)"
> >              String path = store.toString().replace(" (" + store.name() + ")", "");
> > -            // Exclude special directories
> > -            if (path.startsWith("/proc") || path.startsWith("/sys") || path.startsWith("/run") || path.equals("/dev")
> > -                    || path.equals("/dev/pts"))
> > -                continue;
> > +
> > +            // Exclude pseudo file systems
> > +            if (this.pseudofs.contains(store.name())) {
>
> Two things stopping me from merging this is this conditional. One: I don't
> understand what it's doing or why. It may just need some comments to
> explain what's going on, but the three nested if statements are also
> confusing (could be a single if with "or" conditions?
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/dblock/oshi/pull/163/files/0f3b1d30a9e9c494cabd59db647feb7ee12018fe#r62610241>
>